### PR TITLE
Port the publish workflow to GitHub Actions

### LINKED ISSUE #1360: Migrate the current Caliper Azure Pipelines to Github Actions

### LINKED ISSUE #1369: Migrate the publish step to Github Actions
This will require help from the Hyperledger Foundation as they own the npm and docker push secrets, As part of this we will need to also disable the azure pipelines publish to ensure only one build pipeline performs the publish step
This is dependent on #1368 being completed first

### PR DESCRIPTION
**Pull request by CaptainIRS**
In this PR:
* The publish workflow is ported from Azure Pipelines to GitHub Actions as it is. Optimizations would be done in later PRs
* The publish action currently runs after unit and integration tests on committing to the main branch

Closes #1369

This also closes the epic issue for migration.
Closes #1360

### CODE REVIEW

**Review by davidkel — CHANGES_REQUESTED** [review]
In the original azure implementation, when a PR is merged, only the unit tests were run before the publish step rather than the unit and integration tests so it's this we need to replicate for now.

I'm also thinking that to avoid conflict we need to remove the azure pipelines publish at the same time.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  push:
+    branches: [ "main" ]
+
+jobs:
+  unit-tests:
+    uses: ./.github/workflows/unit-tests.yml
+  publish-caliper:
+    uses: ./.github/workflows/publish.yml
+    needs: unit-tests
+  integration-tests:
```
Comment (davidkel) [inline .github/workflows/main.yml:13]: If this is a PR merge then we don't need to run the integration tests. The original azure pipelines just runs unit-tests followed by a publish on a merge

**Review by davidkel — CHANGES_REQUESTED** [review]
Looking good, just a couple of comments and to confirm that I understand the purpose of main.yml

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: CI (main)
```
[the remark this answers is not in the document]
Comment (CaptainIRS) [reply .github/workflows/main.yml:1]: This action is for a push to main (which includes a PR merge). Should I change it to something like "CI (merge)"?

[the remark this answers is not in the document]
Comment (davidkel) [reply .github/workflows/main.yml:1]: How about CI (Publish) as this is what the CI is going to do here ?

**Review by davidkel — APPROVED** [review]
LGTM

**Pull request merged**